intro-to-R-tidyverse/00b-debugging_resources.md

```diff
@@ -0,0 +1,244 @@
+# An Introductory Guide to Troubleshooting Errors
+
+This guide gives you tips and approaches for fixing errors that will arise. We 
+also go through some of the most common errors that you will encounter and what 
+they mean. 
+
+## Tips for approaching errors and bugs
+
+### 1) Carefully read any and all error messages
+
+This may seem like a silly thing to include as a tip, but it's very easy to
+gloss over an error message without actually reading it. Often, R may be telling
+you exactly what is wrong, but if you don't take the time to understand what the
+error message means, you will have trouble fixing the error. Error messages 
+often refer to R terms (e.g.. "argument", "directory") so if you need a refresher 
+on what some terms mean, we recommend running through the [`intro_to_R` module](https://alexslemonade.github.io/training-modules/intro-to-R-tidyverse/01-intro_to_r.nb.html)
+or going through Hadley Wickham's [R for Data Science](https://r4ds.had.co.nz/)
+
+Secondly, realize that just because you don't receive an error message, doesn't 
+mean that your code did what you intended it to. You also will need to carefully
+review your code (and your results) to try to find "silent" bugs (situations 
+where R did exactly what you asked, but you didn't get what you intended).
+
+### 2) Identify which line and phrase of code is the source of the error. 
+
+If you ran many lines of code, you may not know which part of your code is the 
+origin of the error message. Isolating the source of the error and trying to 
+better understand your problem, should be your first course of action. 
+The best way to determine this, is by running each line, and each phrase by 
+itself, one at a time. 
+
+Chunk-out your code and test the individual bits of code. Do you have a lot of 
```

Comment:
Is the term "chunk-out" too jargony?

Reply:
I don't think so. I think the general population uses the word 'chunk' in the same way.